cmd/kube-scheduler/app/server.go

```diff
@@ -167,10 +167,12 @@ func Run(ctx context.Context, cc *schedulerserverconfig.CompletedConfig, sched *
 	defer cc.EventBroadcaster.Shutdown()
 
 	// Setup healthz checks.
-	var checks []healthz.HealthChecker
+	var checks, readyzChecks []healthz.HealthChecker
 	if cc.ComponentConfig.LeaderElection.LeaderElect {
 		checks = append(checks, cc.LeaderElection.WatchDog)
+		readyzChecks = append(readyzChecks, cc.LeaderElection.WatchDog)
 	}
+	readyzChecks = append(readyzChecks, healthz.NewShutdownHealthz(ctx.Done()))
```

Comment:
I think scheduler should not be ready until **Cache Synced**, not the context is done, right? So that means
- for follower, when DelayCacheUntilActive, always not ready
- otherwise, when cache synced, kube-scheduler is readiness.

Reply:
Yes, this PR does implement as commented if I hadn't missed something.

See https://github.com/kubernetes/kubernetes/pull/118148/files/dc193933204b9450273b9705fecd181277cec854#diff-7a5cd3adb38f371f579372a2c901e73675f590515b35c4a8344217a91eff11fdR299, which always checks the start status of informers (return negative when there is any informer not sync'ed)

Reply:
My understanding on livez is that it purely indicates the process is running and not in stuck state. So livez has very simple check logics. This is also consistent with kube-apiserver in only [applying](https://github.com/kubernetes/apiserver/blob/master/pkg/server/healthz.go#L121) the shutdown to readyz